pkg/deploy/strategy/util/util.go

```diff
@@ -45,7 +46,25 @@ func RecordConfigEvent(events record.EventSink, deployment *kapi.ReplicationCont
 		Count:          1,
 		Type:           eventType,
 	}
-	if _, err := events.Create(event); err != nil {
-		glog.Errorf("Could not send event '%#v': %v", event, err)
+	if _, err := client.Events(ref.Namespace).Create(event); err != nil {
+		glog.Errorf("Could not create event '%#v': %v", event, err)
+	}
+}
+
+func RecordConfigWarnings(client kclient.EventNamespacer, rc *kapi.ReplicationController, decoder runtime.Decoder, out io.Writer) {
+	// TODO: we don't set client in tests, in that case no warnings will be recorded
```

Comment:
No need to pollute the code with these checks just because we don't set them in tests. Tests should be using mocked values.